site/components/Modal/index.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { useEffect, FC } from "react";
+import styled from "@emotion/styled";
+import Close from "@mui/icons-material/Close";
+import { Text } from "@klimadao/lib/components";
+
+import * as styles from "./styles";
+
+type ModalWrapperProps = {
+  showModal: boolean;
+};
+
+export const ModalWrapper = styled.div`
+  display: ${(props: ModalWrapperProps) =>
+    props.showModal ? "block" : "none"};
+`;
+
+export interface Props {
+  showModal: boolean;
+  closeOnBackgroundClick?: boolean;
+  title: string;
+  onToggleModal: () => void;
+}
+
+export const Modal: FC<Props> = (props) => {
+  useEffect(() => {
+    if (props.showModal) {
+      document.body.style.overflow = "hidden";
+    } else {
+      document.body.style.overflow = "";
+    }
+  }, [props.showModal]);
```

Comment:
Disables body scroll when modal is open.

Reply:
Good opportunity for a custom hook 🎉 probably need this for our other modals
```ts
useScrollLock(props.showModal);

// or alternative api for manual control
const setScrollLock = useScrollLock();
if (props.showModal) {
  setScrollLock(true);
}
```

Reply:
Funnily enough, all the other modals don't run into the body scroll issue. It's mind boggling!
Let's card this up as a future refactor that anyone can work on :)